lib/src/material/field/typeahead_field.dart

```diff
@@ -703,6 +726,14 @@ class _TypeAheadFieldState<T> extends State<TypeAheadField<T>>
         }
       }
 
+      if (!_effectiveFocusNode!.hasFocus) {
+        
+        setState(() {
+          _showKeyboard = true;
+        
+        });
+      }
+
```

Comment:
we can save on rebuilds here:
```suggestion
      if (!_effectiveFocusNode!.hasFocus && _showKeyboard) {
        setState(() => _showKeyboard = false);
      }
```